sklearn/ensemble/_gb.py

```diff
@@ -305,9 +512,18 @@ def _init_state(self):
 
         self.init_ = self.init
         if self.init_ is None:
-            self.init_ = self._loss.init_estimator()
+            if is_classifier(self):
```

Comment:
Compared to `main`, I think selecting the initial estimator and selecting terminal nodes are worst codewise. Do you think it's worth making subclasses of the loss functions to bring these abstractions back?

```python
class HalfBinomialLossGB(HalfBinomialLoss):
    def init_estimator(...):
        ...
    def update_terminal_regions(...):
        ...

Reply:
I would really like to keep it the way it is.
I hope the code will come closer to `_update_leaves_values` in the HGBT code some day.

Reply:
I tried to make it a bit nicer with https://github.com/scikit-learn/scikit-learn/pull/26278/commits/f1f1ccb16fa31165e8e1644ed8430f27bfcdbf2a, mainly pulling the logic out of the loop over leaves.

Reply:
I agree that [`f1f1ccb` (#26278)](https://github.com/scikit-learn/scikit-learn/pull/26278/commits/f1f1ccb16fa31165e8e1644ed8430f27bfcdbf2a) is a bit nicer. Let's go with that.